web_monitoring/tests/test_html_diff_validity.py

```diff
@@ -66,6 +67,31 @@ def test_html_diff_render_doesnt_move_script_content_into_page_text():
     assert text_only == ''
 
 
+def test_deactivate_deleted_active_elements():
+    '''
+    Method `_deactivate_deleted_active_elements` is used by `html_diff_render`
+    internally to encapsulate `del > script` and `del > style` elements with a
+    `<template class="wm-diff-deleted-inert">` tag. The result for each deleted
+    tag should be like:
+
+    <del class="wm-diff">
+        <template class="wm-diff-deleted-inert">
+            <script src="s2.js">
+            </script>
+        </template>
+    </del>
+    '''
+    a = '''<body>
+           <script src="s1.js"></script>
+           <p>test</p>
+           <script src="s2.js"></script></body>'''
+    b = '''<body><p>test2</p></body>'''
+    result = html_diff_render(a, b)['combined']
+    soup = html5_parser.parse(result, treebuilder='soup', return_root=False)
+    elements = soup.select('del.wm-diff > template.wm-diff-deleted-inert > script')
+    assert len(elements) == 2
```

Comment:
This would not have failed for the initial PR you made (you can try it!) because you’re just checking for how many `script` elements are in the result, not whether they got moved. The easiest thing might be to count how many `template.wm-diff-deleted-inert` elements there are.

Reply:
Oh, I didn't realise that :( 
I've updated my PR according to your suggestion.